sdk/storage/Azure.Storage.Files.DataLake/src/Models/PathPermissions.cs

```diff
@@ -0,0 +1,249 @@
+        /// information on the owner, owning group, and other, but the ACLs may contain more permissions for specific users
+        /// or groups.
+        /// </summary>
+        public bool ExtendedInfoInAcl { get; set; }
+
+        /// <summary>
+        /// Internal empty constructor.
+        /// </summary>
+        public PathPermissions() { }
+
+        /// <summary>
+        /// Public constructor.
+        /// </summary>
+        /// <param name="owner">The path owner's permissions.</param>
+        /// <param name="group">The path group's permissions.</param>
+        /// <param name="other">Permissions for other users.</param>
+        /// <param name="stickyBit">If sticky bit is enabled</param>
+        /// <param name="extendedInfoInAcl">If there is extended info in the ACL</param>
+        public PathPermissions(
+            RolePermissions owner,
+            RolePermissions group,
+            RolePermissions other,
+            bool stickyBit = false,
+            bool extendedInfoInAcl = false)
+        {
+            Owner = owner;
+            Group = group;
+            Other = other;
+            StickyBit = stickyBit;
+            ExtendedInfoInAcl = extendedInfoInAcl;
+        }
+
+        /// <summary>
+        /// Parses a string in octal format to PathPermissions.
+        /// </summary>
+        /// <param name="octalString">Octal string to parse.</param>
+        /// <returns><see cref="PathPermissions"/>.</returns>
+        public static PathPermissions ParseOctal(string octalString)
+        {
+            if (octalString == null)
+            {
+                return null;
+            }
+
+            if (octalString.Length != 4)
+            {
+                throw new ArgumentException("octalString must be 4 characters");
+            }
+
+            var pathPermissions = new PathPermissions();
+
+            if (octalString[0] == '0')
+            {
+                pathPermissions.StickyBit = false;
+            }
+            else if (octalString[0] == '1')
+            {
+                pathPermissions.StickyBit = true;
+            }
+            else
+            {
+                throw new ArgumentException("First digit of octalString must be 0 or 1");
+            }
+
+            pathPermissions.Owner = RolePermissionsExtensions.ParseOctal(octalString[1]);
+            pathPermissions.Group = RolePermissionsExtensions.ParseOctal(octalString[2]);
+            pathPermissions.Other = RolePermissionsExtensions.ParseOctal(octalString[3]);
+
+            return pathPermissions;
+        }
+
+        /// <summary>
+        /// Parses a symbolic string to PathPermissions.
+        /// </summary>
+        /// <param name="symbolicString">String to parse</param>
+        /// <returns><see cref="PathPermissions"/></returns>
+        public static PathPermissions ParseSymbolic(string symbolicString)
```

Comment:
`ParseSymbolic` feels like half a name.  Maybe `ParseSymbolicPermissions`?

Reply:
Or would it be possible to tell if it's symbolic/octal automatically and just do the right thing with a single Parse method?

Reply:
I'm really not a fan of trying to combine the methods into one.  I'd rather rename ParseSymbolicPermissions and ParseOctalPermssions